libvast/builtins/formats/zeek_tsv.cpp

```diff
@@ -0,0 +1,757 @@
+    t = type{string_type{}};
+  else if (zeek_type == "bool")
+    t = type{bool_type{}};
+  else if (zeek_type == "int")
+    t = type{int64_type{}};
+  else if (zeek_type == "count")
+    t = type{uint64_type{}};
+  else if (zeek_type == "double")
+    t = type{double_type{}};
+  else if (zeek_type == "time")
+    t = type{time_type{}};
+  else if (zeek_type == "interval")
+    t = type{duration_type{}};
+  else if (zeek_type == "addr")
+    t = type{ip_type{}};
+  else if (zeek_type == "subnet")
+    t = type{subnet_type{}};
+  else if (zeek_type == "port")
+    // FIXME: once we ship with builtin type aliases, we should reference the
+    // port alias type here. Until then, we create the alias manually.
+    // See also:
+    // - src/format/pcap.cpp
+    t = type{"port", uint64_type{}};
+  if (!t
+      && (zeek_type.starts_with("vector") or zeek_type.starts_with("set")
+          or zeek_type.starts_with("table"))) {
+    // Zeek's logging framwork cannot log nested vectors/sets/tables, so we can
+    // safely assume that we're dealing with a basic type inside the brackets.
+    // If this will ever change, we'll have to enhance this simple parser.
+    auto open = zeek_type.find('[');
+    auto close = zeek_type.rfind(']');
+    if (open == std::string::npos or close == std::string::npos)
+      return caf::make_error(ec::format_error, "missing container brackets:",
+                             std::string{zeek_type});
+    auto elem = parse_type(zeek_type.substr(open + 1, close - open - 1));
+    if (!elem)
+      return elem.error();
+    // Zeek sometimes logs sets as tables, e.g., represents set[string] as
+    // table[string]. In VAST, they are all lists.
+    t = type{list_type{*elem}};
+  }
+  if (!t)
+    return caf::make_error(ec::format_error,
+                           "failed to parse type: ", std::string{zeek_type});
+  return t;
+}
+
+struct zeek_metadata {
+  using iterator_type = std::string_view::const_iterator;
+
+  auto is_unset(std::string_view field) -> bool {
+    return std::equal(unset_field.begin(), unset_field.end(), field.begin(),
+                      field.end());
+  }
+
+  auto is_empty(std::string_view field) -> bool {
+    return std::equal(empty_field.begin(), empty_field.end(), field.begin(),
+                      field.end());
+  }
+
+  auto parse_header(generator<std::optional<std::string_view>>::iterator it,
+                    const generator<std::optional<std::string_view>>& lines,
+                    operator_control_plane& ctrl) -> caf::expected<void> {
+    auto sep_parser
+      = "#separator" >> ignore(+(parsers::space)) >> +(parsers::any);
+    auto sep_option = std::string{};
+    if (not sep_parser((*it).value(), sep_option)
+        or not sep_option.starts_with("\\x")) {
+      return caf::make_error(
+        ec::syntax_error, fmt::format("Invalid #separator option while parsing "
+                                      "Zeek TSV file - aborting"));
```

Comment:
```suggestion
                                      "Zeek TSV file"));
```

The outside context should probably decide what's going to happen.

Reply:
This comment applies to all occurrences of this suffix.